apps/test/unit/gamelab/animationListModuleTest.js

```diff
@@ -82,4 +86,154 @@ describe('animationListModule', function () {
       expect(store.getState().animationList.propsByKey[key].sourceSize).not.to.be.undefined;
     });
   });
+
+  describe('action: set initial animationList', function () {
+    let oldWindowDashboard, server;
+    beforeEach(function () {
+      oldWindowDashboard = window.dashboard;
+      window.dashboard = {
+        project: {
+          getCurrentId() {return '';}
+        }
+      };
+
+      server = sinon.fakeServer.create();
+      server.respondWith('imageBody');
+    });
+
+    afterEach(function () {
+      server.restore();
+      window.dashboard = oldWindowDashboard;
+    });
+
+    it('when animationList has 1 item, selectedAnimation should be the animation', function () {
+      const key0 = 'anim0';
+      let animationList = {
+        orderedKeys: [key0],
+        propsByKey: {
+          [key0]: {
+            name: 'New animation',
+            sourceUrl: null,
+            frameSize: {x: 100, y: 100},
+            frameCount: 1,
+            looping: true,
+            frameDelay: 4,
+            version: null,
+            loadedFromSource: true,
+            saved: false,
+            blob: null,
+            dataURI: null,
+            hasNewVersionThisSession: false
```

Comment:
Some of these properties wouldn't actually get passed to `setInitialAnimationList` because they belong to [`AnimationProps`](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/gamelab/PropTypes.js#L95), not [`SerializedAnimationProps`](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/gamelab/PropTypes.js#L61).  Your test will be a little more accurate if you only pass serialized props to this method.